### PR TITLE
Deploy from GIT url

### LINKED ISSUE #514: Deploy from Git URL
We can create an app with a Git repo (creds maybe) and push it without uploading a tarball first.
The Git info is stored in or referenced from the application resource.

might rely on #513

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fixes #514

### CODE REVIEW

File: internal/cli/push.go [hunk 1]
```diff
@@ -83,8 +90,18 @@ var CmdPush = &cobra.Command{
...
 
+		gitRevision, err := cmd.Flags().GetString("git")
+		if err != nil {
+			return errors.Wrap(err, "could not read option --git")
+		}
+
 		var path string
 		if len(args) == 1 {
+			if gitRevision != "" {
+				cmd.SilenceUsage = false
```
Comment (jimmykarily) [inline internal/cli/push.go:101]: I think you need to unsilence in this case too:

```
./dist/epinio-linux-amd64 push --git main https://github.com/epinio/example-12factor.git
```
This is missing the app name but it prints no error, just exits.

Comment (andreas-kupries) [reply internal/cli/push.go:101]: The code here should be unsilencing things. Double Negation: silence = false => do not silence = unsilence.
Will investigate.

Comment (andreas-kupries) [reply internal/cli/push.go:101]: Hm. removing the statement makes no difference. No output ... disabling the `silence = true` at the top ... no difference ... I am beginning to hate cobra error handling.

Comment (andreas-kupries) [reply internal/cli/push.go:101]: :bulb: I am using errors.Wrap, except, the `err` is likely nil :scream:

**Comment by jimmykarily** [comment]
Can we add docs for this? Or should we do that here: https://github.com/epinio/epinio/issues/377 ? I don't want us to start delegating documentation to that story but given it's already in progress, maybe we will get unnecessary conflicts or even loose docs on rebases and such. I'm ok if you just add a commit to the wip branch of #377 to document this.

File: internal/cli/push.go (hunk 1 above)
Comment (andreas-kupries) [reply internal/cli/push.go:107]: Fixed with commit b526fc85e7baa8906a9444a9b828f1d36ff67625

**Comment by andreas-kupries** [comment]
I will do the doc in #377

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**